pareto/strategic_water_management/strategic_produced_water_optimization.py

```diff
@@ -3405,6 +3494,350 @@ def SeismicActivityExceptionRule(model, k, t):
+        initialize=130,
+        mutable=True,
+        units=pyunits.dimensionless,
+        doc="Pipeline material factor for Hazen-Williams equation",
+    )
+    mh.v_PumpHead = Var(
+        model.s_LLA,
+        model.s_T,
+        within=NonNegativeReals,
+        initialize=0,
+        units=pyunits.meter,
+        doc="Pump Head added in the direction of flow, m",
+    )
+    mh.v_ValveHead = Var(
+        model.s_LLA,
+        model.s_T,
+        within=NonNegativeReals,
+        initialize=0,
+        units=pyunits.meter,
+        doc="Valve Head removed in the direction of flow, m",
+    )
+    mh.p_rhog = Param(
+        initialize=9.8 * 1000,
+        units=model.model_units["pressure"] / pyunits.meter,
+        doc="g (m/s2) * density of PW (assumed to be 1000 kg/m3)",
+    )
+    mh.p_PumpingCost = Param(
+        initialize=1.2,
+        doc="Cost of pumping water",
+    )
+    mh.p_Min_AOP = Param(
+        initialize=pyunits.convert_value(
+            model.df_parameters["Hydraulics"]["min_allowable_pressure"],
+            from_units=model.user_units["pressure"],
+            to_units=model.model_units["pressure"],
+        ),
+        mutable=True,
+        units=model.model_units["pressure"],
+        doc="Minimum ALlowable Operating Pressure",
+    )
+    mh.p_Max_AOP = Param(
+        initialize=pyunits.convert_value(
+            model.df_parameters["Hydraulics"]["max_allowable_pressure"],
+            from_units=model.user_units["pressure"],
+            to_units=model.model_units["pressure"],
+        ),
+        mutable=True,
+        units=model.model_units["pressure"],
+        doc="Maximum ALlowable Operating Pressure",
+    )
+    mh.p_WellPressure = Param(
+        model.s_P,
+        model.s_T,
+        default=0,
+        initialize={
+            key: pyunits.convert_value(
+                value,
+                from_units=model.user_units["pressure"],
+                to_units=model.model_units["pressure"],
+            )
+            for key, value in model.df_parameters["WellPressure"].items()
+        },
+        units=model.model_units["pressure"],
+        doc="Well pressure at production or completions pad [pressure]",
+    )
+    mh.v_Pressure = Var(
+        model.s_L,
+        model.s_T,
+        within=NonNegativeReals,
+        initialize=mh.p_Min_AOP,
+        bounds=(mh.p_Min_AOP, None),
+        units=model.model_units["pressure"],
+        doc="Pressure at location l at time t in Pa",
+    )
+    for p in model.s_P:
+        for t in model.s_T:
+            if value(mh.p_WellPressure[p, t]) > 0:
+                mh.v_Pressure[p, t].fix(mh.p_WellPressure[p, t])
+
+    def MAOPressureRule(b, l1, t1):
+        constraint = b.v_Pressure[l1, t1] <= b.p_Max_AOP
+        return process_constraint(constraint)
+
+    mh.MAOPressure = Constraint(
+        model.s_L - model.s_P,
+        model.s_T,
+        rule=MAOPressureRule,
+        doc="Max allowable pressure rule",
+    )
```

Comment:
I don't see a corresponding constraint for min allowable pressure - is that something that needs to be added?

Reply:
the min allowable pressure is added as a bound but the max allowable pressure is added as a constraint to allow the flexibility to change the value during the analysis

Reply:
in addition to the aforementioned comment, the MaxAOP is not added as a bound because it is only enforced for some nodes. For example, the production pads and completion pads are not subjected to this constraint.

Reply:
Thanks for the explanation; resolving.